app/models/sale.rb

```diff
@@ -0,0 +1,5 @@
+class Sale < ActiveRecord::Base
+  belongs_to :vendor
+  validates :amount, presence: true, numericality: {only_integer: true }
```

Comment:
Nice use of validation here to make sure you are getting the type of data you expect.